test/integration/quic_http_integration_test.cc

```diff
@@ -1167,6 +1167,86 @@ TEST_P(QuicHttpIntegrationTest, MultipleNetworkFilters) {
   codec_client_->close();
 }
 
+TEST_P(QuicHttpIntegrationTest, DeferredLogging) {
+  useAccessLog("%PROTOCOL%,%ROUNDTRIP_DURATION%");
+  initialize();
+  codec_client_ = makeHttpConnection(makeClientConnection(lookupPort("http")));
+  sendRequestAndWaitForResponse(default_request_headers_, /*request_size=*/0,
+                                default_response_headers_,
+                                /*response_size=*/0,
+                                /*upstream_index=*/0, TestUtility::DefaultTimeout);
+  codec_client_->close();
+
+  std::string log = waitForAccessLog(access_log_name_);
+  std::string delimiter = ",";
+  std::string protocol = log.substr(0, log.find(delimiter));
+  std::string roundtrip_duration = log.substr(log.find(delimiter) + 1, log.length());
+  EXPECT_EQ(protocol, "HTTP/3");
+  EXPECT_GT(std::stoi(roundtrip_duration), 0);
+}
+
+TEST_P(QuicHttpIntegrationTest, DeferredLoggingDisabled) {
+  config_helper_.addRuntimeOverride("envoy.reloadable_features.quic_defer_logging_to_ack_listener",
+                                    "false");
+  useAccessLog("%PROTOCOL%,%ROUNDTRIP_DURATION%");
+  initialize();
+  codec_client_ = makeHttpConnection(makeClientConnection(lookupPort("http")));
+  sendRequestAndWaitForResponse(default_request_headers_, /*request_size=*/0,
+                                default_response_headers_,
+                                /*response_size=*/0,
+                                /*upstream_index=*/0, TestUtility::DefaultTimeout);
+  codec_client_->close();
+
+  // Do not flush client acks.
+  std::string log = waitForAccessLog(access_log_name_, 0, false, nullptr);
+  std::string delimiter = ",";
+  std::string protocol = log.substr(0, log.find(delimiter));
+  std::string roundtrip_duration = log.substr(log.find(delimiter) + 1, log.length());
+  EXPECT_EQ(protocol, "HTTP/3");
+  EXPECT_EQ(roundtrip_duration, "-");
+}
+
+TEST_P(QuicHttpIntegrationTest, DeferredLoggingWithReset) {
+  useAccessLog("%PROTOCOL%,%ROUNDTRIP_DURATION%");
```

Comment:
please also check that other metrics are logged.

Reply:
Added several more metrics.